system/jlib/jarray.hpp

```diff
@@ -440,7 +439,7 @@ class ReferenceArrayMapper
     {
         member = &newValue;
     }
-    static inline void destruct(ITEM * & member)
+    static inline void destruct(ITEM * & member __attribute((unused)))
```

Comment:
IMO, it is not worth introducing hiding behind a macro.

It would be if different compilers needed different definitions (or none).